### PR TITLE
Fix datetime unmarshalling errors

### PR DESCRIPTION
**Pull request by mgleung**
## Description
Fixes some of the failing STs regarding python timestamps being in an invalid format.

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: calico_node/tests/st/policy/test_felix_gateway.py [hunk 1]
```diff
@@ -844,7 +844,12 @@ def delete_all(self, resource):
             if 'items' in objects and len(objects['items']) == 0:
                 pass
             else:
-                self._delete_data(objects, self.hosts[0])
+                # clean out the incorrectly formatted python datetime timestamps
+                if 'items' in objects:
+                    for obj in objects['items']:
+                        if 'metadata' in obj and isinstance(obj['metadata'], dict):
+                            obj['metadata'].pop('creationTimestamp')
+                    self._delete_data(objects, self.hosts[0])
```
Comment (bcreane) [inline calico_node/tests/st/policy/test_felix_gateway.py]: this should not be under the if statement, we should delete_data irrespective of items in objects ....

**Comment by mgleung** [comment]
@caseydavenport this will be needed in addition to the libcalico-go version update. This fixes python's datetime object not handling timezone information by default.

**Comment by bcreane** [comment]
looks like libcalico-go pin didn't pick up @caseydavenport 's recent fix.

**Comment by neiljerram** [comment]
@caseydavenport To expand on @mgleung 's point above, both ingredients here are needed, namely:

1. We need the libcalico-go bump so that non-null creationTimestamps are present on the resources that calico/node creates at start of day, in particular IP Pools.

2. We need ST code hackery to delete creationTimestamp fields when applying/updating/deleting resources - when we've unmarshaled and remarshaled via a Python dict - because the Python marshaling roundtrip unfortunately transforms a syntactically valid timestamp:
```
    creationTimestamp: 2017-11-07T12:49:49Z
```
to a syntactically invalid one:
```
    creationTimestamp: 2017-11-07 12:49:49
```

**Comment by neiljerram** [comment]
I now also have those ingredients in my PR at #1319 - thanks @mgleung for the inspiration!  If I get ahead enough to be able to merge that one, we won't actually need this PR here.

**Comment by neiljerram** [comment]
Subsumed within #1319

**Pull request closed without merging**